src/main/com/wsscode/pathom3/connect/planner.cljc

```diff
@@ -1091,42 +1106,44 @@
     ::pco/keys    [op-name]
     ast           :edn-query-language.ast/node
     :as           env}]
-  (let [ast-params (:params ast)
-        config     (pci/resolver-config env op-name)
-        op-name'   (or (::pco/dynamic-name config) op-name)
-        dynamic?   (pci/dynamic-resolver? env op-name')
-        sub        (if dynamic? (compute-dynamic-nested-requirements env))
-        requires   {attribute (or sub {})}]
-    (cond->
-      (new-node env
-                {::pco/op-name op-name'
-                 ::expects     requires
-                 ::input       input})
-
-      (seq ast-params)
-      (assoc ::params ast-params)
-
-      dynamic?
-      (assoc
-        ::source-op-name
-        op-name
-
-        ::foreign-ast
-        {:type     :root
-         :children (if sub
-                     (let [ast' (pfsd/shape-descriptor->ast sub)]
-                       [(assoc ast
-                          :children (:children ast')
-                          :query (pfsd/shape-descriptor->query sub))])
-                     [ast])}))))
+  (let [{:keys [::pco/provides] :as config} (pci/resolver-config env op-name)
+        resolver-provided-shape             (get provides (:key ast))
+        ast-required-shape                  (pfsd/ast->shape-descriptor ast)]
+    (when-not (shape-statically-unreachable? env resolver-provided-shape ast-required-shape)
```

Comment:
here is important to check if both the nested query AND the resolver sub-query have details (something not empty inside). If any is blank (the sub-query or the resolver attribute output subquery) we should short-circuit and accept the node.

Reply:
This check is done in `shape-statically-unreachable?`. I'll move it here and get rid of the extra function.

Reply:
I may well have misunderstood things but it seemed like there is behaviour that relies on `shape-unreachable?` throwing in order to short circuit processing. Again, let me investigate a bit more and I'll get back to you.